R/sugar.R

```diff
@@ -66,3 +66,157 @@ opt = function(.key, ...) {
 opts = function(.keys, ...) {
   dictionary_sugar_mget(mlr_optimizers, .keys, ...)
 }
+
+
+#' @title Black-Box Optimization
+#'
+#' @description
+#' This function optimizes a function or [Objective] with a given method.
+#'
+#' @param method (`character(1)`)\cr
+#'  Key to retrieve optimizer from [mlr_optimizers] dictionary.
+#' @param fun (`function`)\cr
+#'  Objective function to be minimized. If the function should be maximized, a
+#'  `codomain` tagged with `maximize` must be set. Either `fun` or `objective`
+#'  must be given.
+#' @param objective ([Objective])\cr
+#'  Objective. Either `fun` or `objective` must be given.
+#' @param term_evals (`integer(1)`)\cr
+#'  Number of allowed evaluations.
+#' @param term_time (`integer(1)`)\cr
+#'  Maximum allowed time in seconds.
+#' @param search_space ([paradox::ParamSet])\cr
+#'  Search space. If `fun` is used, either `search_space` or `lower` and `upper`
+#'  must be given.
+#' @param lower (`numeric()`)\cr
+#'  Lower bounds on the parameters. If named, names are used to create the
+#'  search space. If `fun` is used, either `search_space` or  `lower` and
+#'  `upper` must be given.
+#' @param upper (`numeric()`)\cr
+#'  Upper bounds on the parameters. If `fun` is used, either `search_space` or 
+#' `lower` and `upper` must be given.
+#' @param codomain ([paradox::ParamSet])\cr
+#'  Optional codomain. If not given, an unbounded double parameter `y` is
+#'  created.
+#' @param ... (named `list()`)\cr
```

Comment:
Should perhaps be passed on to objective function. If the user wants to have special args to the optimizer, the user can also just provide the optimizer?

Reply:
or possibly pass additional opt arguments as a `control` argument; `optimizer = invoke(opt, method, .args = control)`

Reply:
changing my mind about this; maybe `method` should be an `Optimizer` method and the user should just call `bboptim(fun, opt(<method>))`; it is not that much more to type.

Reply:
Passed to the objective function now.